File: gofer/src/api/fmt.rs
use gofer_models::task_run;

pub fn task_run_log_path(log_dir: &str, task_run: &task_run::TaskRun) -> String {
    return format!(
        "{}/{}_{}_{}_{}",
        log_dir, &task_run.namespace, task_run.pipeline, task_run.run, task_run.id
    );
}

pub fn secret_key(namespace: &str, pipeline: &str, key: &str) -> String {
    return format!("{}_{}_{}", namespace, pipeline, key);
}

pub fn pipeline_object_key(namespace: &str, pipeline: &str, key: &str) -> String {
    return format!("{}_{}_{}", namespace, pipeline, key);
}

pub fn run_object_key(namespace: &str, pipeline: &str, run: u64, key: &str) -> String {
    return format!("{}_{}_{}_{}", namespace, pipeline, run, key);
}

pub fn task_container_id(namespace: &str, pipeline: &str, run: u64, task_run: &str) -> String {
    return format!("{}_{}_{}_{}", namespace, pipeline, run, task_run);
}

pub fn trigger_container_id(name: &str) -> String {
    return format!("trigger_{}", name);
}

File: internal/api/fmt.go
package api

import (
	"fmt"
)

const (
	ObjectPipelineKeyFmt    = "%s_%s_%s"         // <namespaceid>_<pipelineid>_<key>
	ObjectRunKeyFmt         = "%s_%s_%d_%s"      // <namespaceid>_<pipelineid>_<runid>_<key>
	GlobalSecretKeyFmt      = "global_secret_%s" // global_secret<key>
	PipelineSecretKeyFmt    = "%s_%s_%s"         // <namespaceid>_<pipelineid>_<key>
	TaskContainerIDFmt      = "%s_%s_%d_%s"      // <namespaceid>_<pipelineid>_<runid>_<taskrunid>
	ExtensionContainerIDFmt = "extension_%s"     // extension_<name>
	InstallerContainerIDFmt = "installer_%s"     // installer_<randomly-generated-value>
	TaskRunFilePath         = "%s/%s_%s_%d_%s"   // folder/<namespaceid>_<pipelineid>_<runid>_<taskrunid>
)

func globalSecretKey(key string) string {
	return fmt.Sprintf(GlobalSecretKeyFmt, key)
}

func pipelineSecretKey(namespace, pipeline, key string) string {
	return fmt.Sprintf(PipelineSecretKeyFmt, namespace, pipeline, key)
}

func pipelineObjectKey(namespace, pipeline, key string) string {
	return fmt.Sprintf(ObjectPipelineKeyFmt, namespace, pipeline, key)
}

func runObjectKey(namespace, pipeline string, run int64, key string) string {
	return fmt.Sprintf(ObjectRunKeyFmt, namespace, pipeline, run, key)
}

func taskContainerID(namespace, pipeline string, run int64, taskRun string) string {
	return fmt.Sprintf(TaskContainerIDFmt, namespace, pipeline, run, taskRun)
}

func extensionContainerID(name string) string {
	return fmt.Sprintf(ExtensionContainerIDFmt, name)
}

func installerContainerID() string {
	return fmt.Sprintf(InstallerContainerIDFmt, generateToken(5))
}

func taskRunLogFilePath(dir, namespace, pipeline string, run int64, task string) string {
	return fmt.Sprintf(TaskRunFilePath, dir, namespace, pipeline, run, task)
}
